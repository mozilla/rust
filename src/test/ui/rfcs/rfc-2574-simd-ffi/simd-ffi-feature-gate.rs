// only-x86_64

#![feature(repr_simd)]
#![feature(avx512_target_feature)]
#![allow(non_camel_case_types)]
#![cfg(any(target_arch = "x86", target_arch = "x86_64"))]

#[repr(simd)]
struct v128(i128);

extern {
    fn foo(x: v128); //~ ERROR use of SIMD type `v128` in FFI is unstable
    fn bar(x: i32, y: v128); //~ ERROR use of SIMD type `v128` in FFI is unstable
    fn baz(x: i32) -> v128; //~ ERROR use of SIMD type `v128` in FFI is unstable
}

fn main() {}
