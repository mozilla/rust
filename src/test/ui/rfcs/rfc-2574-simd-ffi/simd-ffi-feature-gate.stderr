error: use of SIMD type `v128` in FFI is unstable
  --> $DIR/simd-ffi-feature-gate.rs:12:15
   |
LL |     fn foo(x: v128);
   |               ^^^^
   |
   = help: add `#![feature(simd_ffi)]` to the crate attributes to enable

error: use of SIMD type `v128` in FFI is unstable
  --> $DIR/simd-ffi-feature-gate.rs:13:23
   |
LL |     fn bar(x: i32, y: v128);
   |                       ^^^^
   |
   = help: add `#![feature(simd_ffi)]` to the crate attributes to enable

error: use of SIMD type `v128` in FFI is unstable
  --> $DIR/simd-ffi-feature-gate.rs:14:23
   |
LL |     fn baz(x: i32) -> v128;
   |                       ^^^^
   |
   = help: add `#![feature(simd_ffi)]` to the crate attributes to enable

error: aborting due to 3 previous errors

